peermanager/peermanager.go

```diff
@@ -114,3 +114,14 @@ func (pm *PeerManager) getOrCreate(p peer.ID) *peerProcessInstance {
 	}
 	return pqi
 }
+
+func (pm *PeerManager) onQueueShutdown(p peer.ID) {
+	pm.peerProcessesLk.Lock()
+	defer pm.peerProcessesLk.Unlock()
+	_, ok := pm.peerProcesses[p]
+	if !ok {
+
+		return
+	}
```

Comment:
Remove the code that checks for the presence of `p`. It is not needed as `delete` already does that.

Reply:
right.